data/src/main/java/org/apache/iceberg/data/GenericDeleteFilter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.data;
+
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.InputFile;
+
+public class GenericDeleteFilter extends DeleteFilter<Record> {
+  private final FileIO io;
+  private final InternalRecordWrapper asStructLike;
+
+  public GenericDeleteFilter(FileIO io, FileScanTask task, Schema tableSchema, Schema requestedSchema) {
+    super(task, tableSchema, requestedSchema);
+    this.io = io;
+    this.asStructLike = new InternalRecordWrapper(requiredSchema().asStruct());
+  }
+
+  @Override
+  protected long pos(Record record) {
+    return (Long) posAccessor().get(record);
+  }
+
+  @Override
+  protected StructLike asStructLike(Record record) {
+    return asStructLike.wrap(record);
```

Comment:
Reconsidered this again,  I saw argument `record` is actually read from the `requiredSchema`,  do we need to wrap this record by the same `requiredSchema` again  ?  Pls see https://github.com/apache/iceberg/pull/1444/files#diff-a8a025276b1d93b0830f2ee6c91118efR76.

If the record is actually matching the `requiredSchema`,  then we  could just return the `record` in `asStructLike` method  ?  Also we don't have to overwrite `pos` method in this classes again ?

Reply:
The purpose of the wrapper here is to translate from Iceberg's generic to the internal representation for values. For example, generics will pass timestamptz as an `OffsetDateTime` to users, but internally Iceberg uses microseconds from epoch as a long.